### PR TITLE
[docs] Update release notes for APX relocation types

### PR DESCRIPTION
**Pull request by fzou1**

### CODE REVIEW

**Comment by fzou1** [comment]
It depends on #117675. Please review after that PR is merged. Thanks.

File: lld/docs/ReleaseNotes.rst [hunk 1]
```diff
@@ -35,6 +35,11 @@ ELF Improvements
   globally changing the semantics of section matching. It also independently
   increases the expressive power of linker scripts.
   (`#95323 <https://github.com/llvm/llvm-project/pull/95323>`_)
+* Supported relocation types for X86_64 target:
+  * ``R_X86_64_CODE_4_GOTPCRELX`` (`#109783 <https://github.com/llvm/llvm-project/pull/109783>`_ and `#116737 <https://github.com/llvm/llvm-project/pull/116737>`_)
```
Comment (phoebewang) [inline lld/docs/ReleaseNotes.rst:39]: Should we use
```
(`#109783 <https://github.com/llvm/llvm-project/pull/109783>`_) and (`#116737 <https://github.com/llvm/llvm-project/pull/116737>`_)
```

Comment (MaskRay) [reply lld/docs/ReleaseNotes.rst:39]: X86_64 => x86-64

Comment (fzou1) [reply lld/docs/ReleaseNotes.rst]: Updated.

**Review by phoebewang — APPROVED** [review]
LGTM.

**Review by MaskRay — APPROVED** [review]

**Review by KanRobert — APPROVED** [review]

**Pull request merged**